File: services/nuldoc/nuldoc-src/generators/about.ts
import AboutPage from "../pages/AboutPage.ts";
import { Config } from "../config.ts";
import { Page } from "../page.ts";
import { SlidePage } from "./slide.ts";

export type AboutPage = Page;

export async function generateAboutPage(
  slides: SlidePage[],
  config: Config,
): Promise<AboutPage> {
  const html = await AboutPage(slides, config);

  return {
    root: html,
    renderer: "html",
    site: "about",
    destFilePath: "/index.html",
    href: "/",
  };
}

File: services/nuldoc/lib/nuldoc/generators/about.rb
module Nuldoc
  module Generators
    class About
      def initialize(slides, staff_records, config)
        @slides = slides
        @staff_records = staff_records
        @config = config
      end

      def generate
        html = Pages::AboutPage.new(slides: @slides, staff_records: @staff_records, config: @config).render

        Page.new(
          root: html,
          renderer: :html,
          site: 'about',
          dest_file_path: '/index.html',
          href: '/'
        )
      end
    end
  end
end
